src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -290,23 +290,33 @@ public void SslStream_StreamToStream_Write_ReadByte_Success()
             VirtualNetwork network = new VirtualNetwork();
 
             using (var clientStream = new VirtualNetworkStream(network, isServer: false))
-            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var serverStream = new NotifyReadVirtualNetworkStream(network, isServer: true))
             using (var clientSslStream = new SslStream(clientStream, false, AllowAnyServerCertificate))
             using (var serverSslStream = new SslStream(serverStream))
             {
                 bool result = DoHandshake(clientSslStream, serverSslStream);
                 Assert.True(result, "Handshake completed.");
 
-                byte serverReadByte = 0;
-                var readTask = Task.Run(async () =>
+                var serverBuffer = new byte[1];
+                var tcs = new TaskCompletionSource<object>();
+                serverStream.OnRead += (buffer, offset, count) =>
                 {
-                    var serverBuffer = new byte[1];
-                    await serverSslStream.ReadAsync(serverBuffer, 0, serverBuffer.Length);
-                    serverReadByte = serverBuffer[0];
-                });
+                    if (!tcs.Task.IsCompleted)
+                    {
+                        tcs.SetResult(null);
+                    }
+                };
+                var readTask = serverSslStream.ReadAsync(serverBuffer, 0, serverBuffer.Length).ConfigureAwait(false);
+
+                // Since the sequence of calls that ends in serverStream.Read() is sync, by now
+                // the read task will have acquired the semaphore shared by Stream.BeginReadInternal()
+                // and Stream.BeginWriteInternal().
+                // But to be sure, we wait until we know we're inside Read().
+                await tcs.Task.TimeoutAfter(TestConfiguration.PassingTestTimeoutMilliseconds).ConfigureAwait(false);
```

Comment:
ConfigureAwait(false) generally isn't necessary in our tests, just in the product source